BridgeAppSDK/SBABrainBaselineStepViewController.swift

```diff
@@ -0,0 +1,229 @@
+        else {
+            testDidFinish(result: nil)
+            return
+        }
+        self.startDate = Date()
+        self.show(viewController, sender: self)
+    }
+    
+    open func createTestViewController() -> UIViewController? {
+        // Subclass must override to implement methods on private framework.
+        assertionFailure("Abstract method")
+        return nil
+    }
+    
+    open override func show(_ vc: UIViewController, sender: Any?) {
+        addChildViewController(vc)
+        vc.view.frame = self.view.bounds
+        view.addSubview(vc.view)
+        vc.didMove(toParentViewController: self)
+    }
+    
+    open func testDidFinish(result: Any?) {
+        if result == nil {
+            // If the user quit the brain baseline task then tell the task view controller delegate that
+            // the result was discarded.
+            self.taskViewController!.delegate?.taskViewController(self.taskViewController!, didFinishWith: .discarded, error: nil)
+        }
+        else {
+            self.goForward()
+        }
+    }
+}
+
+extension SBAUserWrapper {
+    
+    /**
+     Returns a unique identifier associated with this user.
+     */
+    public func uniqueIdentifier() -> String {
+        
+        // Look to see if there is a stored answer for a unique identifier that is
+        // *not* the hashed email address. If so, return that. This is not defined
+        // using a guard statement b/c we want an early exit if and only if the
+        // identifier is already defined.
+        let uniqueIdentifierUserKey = "uniqueIdentifier"
+        if let uuid = self.storedAnswer(for: uniqueIdentifierUserKey) as? String {
+            return uuid
+        }
+        
+        // Create a unique identifier by hashing the email address (if available)
+        guard let email = self.email, let data = email.data(using: .utf8) as NSData?, let md5 = data.hexMD5()
+            else {
+                // If the email cannot be hashed to create a unique identifier, then create and store a uuid
+                let uuid = UUID().uuidString
+                self.setStoredAnswer(uuid, forKey: uniqueIdentifierUserKey)
+                return uuid
+        }
+        
+        // If the md5 was created then return that
+        return md5
+    }
+    
+    /**
+     Mapping of key/value pairs for
+     */
+    public func brainBaselineProperties() -> [String: Any] {
+
+        // if user has consented we will have their birthdate (at least the year).
+        let year = Calendar.current.component(.year, from: self.birthdate!)
+        
+        let gender = SBAProfileManager.shared?.value(forProfileKey: SBAProfileInfoOption.gender.rawValue)
+        let genderString: String? = (gender as? String) ?? ((gender as? HKBiologicalSex)?.demographicDataValue as String?)
+        
+        // TODO: emm2017-04-30 Set other demographic values we want to compare against, e.g. education
+        
+        return ["birth_year" : year,
```

Comment:
the list of demographic attributes the data can be grouped by will be different for different studies, so maybe in a protocol extension isn't ideal for this?